### DISCUSSION TITLE
Adding font-style and text-transform utilities?

### DISCUSSION THREAD

**Discussion by bakura10** [tailwindlabs/tailwindcss#7864] [Ideas]
Hi,

As of today Tailwind offers built-in utililities like "italic" or "normal", but this makes it impossible to use CSS variables when the font style or text-transform is driven by a user setting that changes a CSS variable.

This is not a _super huge_ deal, but it forces to do that:

```
.foo {
  @apply text-h1 font-heading leading-heading;
  font-style: var(--heading-font-style);
  text-transform: var(--heading-text-transform);
}
```

While this would be more consistent:

```
.foo {
  @apply text-h1 font-heading leading-heading case-heading style-heading;
}
```

**Comment by bakura10** [comment]
It's always when I post the question that I find a solution:

`@apply font-heading tracking-heading [font-style:var(--heading-font-style)] [text-transform:var(--heading-text-transform)];`

I think this therefore can be closed :).

**Comment by ZeldOcarina** [comment]
This was actually a nice point. Where is this documented? Also, this cannot be customized in tailwind config so I think it may be worth some improvements.
